packages/hardhat-chai-matchers/src/internal/changeTokenBalance.ts

```diff
@@ -76,12 +89,14 @@ export function supportChangeTokenBalance(Assertion: Chai.AssertionStatic) {
       );
       const addressesPromise = Promise.all(accounts.map(getAddressOf));
 
-      const derivedPromise = Promise.all([
-        balanceChangesPromise,
-        addressesPromise,
-        getTokenDescription(token),
-      ]).then(([actualChanges, addresses, tokenDescription]) => {
-        this.assert(
+      const checkBalanceChanges = ([
+        actualChanges,
+        addresses,
+        tokenDescription,
+      ]: [EthersT.BigNumber[], string[], string]) => {
```

Comment:
Just out of curiosity: why does this instance of `checkBalanceChanges` have a return type while the others don't?

Reply:
I'm not sure I understand. The other `check*` functions have a similar type too. Notice though that this isn't the return type, it's the type of the parameters of the function, and the reason I had to add it is that it can no longer be inferred because it's not used directly as an argument anymore. Does that make sense?